### DISCUSSION TITLE
How to automatic delete background?

### DISCUSSION THREAD

**Discussion by Silverhawk85** [T8RIN/ImageToolbox#599] [General]
Hi,

I like to know how the automatic function for deleting the background works. I couldn't figure it out.

Regards
Markus

**Comment by ebarruel** [comment]
I also can't figure it out (on Android)

**Comment by T8RIN** [comment]
In foss version it doesn't exist

**Comment by maruuk** [comment]
Will this ever be possible in FOSS version? It would be great to have this feature in the app.

**Reply by T8RIN** [reply]
Nope, unreal, it uses google mlkit to work

**Reply by ebarruel** [reply]
Oh okay, I misunderstood what "auto" meant in this case. I thought there was a way to select a certain area and delete what was inside or around it.

**Reply by T8RIN** [reply]
It's possible

**Reply by T8RIN** [reply]
> Oh okay, I misunderstood what "auto" meant in this case. I thought there was a way to select a certain area and delete what was inside or around it.

You can use mask filters to achieve this behavior, but it's a bit not obvious

Create a mask, selct inverse fill type and select replace color filter, then use color replacement to transparent, and tolerance at the full value, this will strip image around mask, you can save it and viola, but transparent parts will be not stripped using this way :( so maybe i'll add this as an option to existing image eraser to be more understandable and usable

**Reply by ebarruel** [reply]
I didn't even think of that. Thanks!
